river/stream/iter_pandas.py

```diff
@@ -40,9 +46,15 @@ def iter_pandas(
     {'x1': 4, 'x2': 'blue'} True
 
     """
-
-    kwargs["feature_names"] = X.columns
-    if isinstance(y, pd.DataFrame):
-        kwargs["target_names"] = y.columns
-
-    yield from stream.iter_array(X=X.to_numpy(), y=y if y is None else y.to_numpy(), **kwargs)
+    msg = "Please use `stream.iter_frame`. `stream.iter_pandas` is deprecated, and it will be removed in future versions."
+    warn(msg, DeprecationWarning)
+
+    if (pd := nw.dependencies.get_pandas()) is not None:
+        if not isinstance(X, pd.DataFrame):
+            msg = f"Expected pandas DataFrame, received {type(X)}"
+            raise TypeError(msg)
+        if y is not None and not isinstance(y, (pd.DataFrame, pd.Series)):  # noqa: UP038
+            msg = f"Expected pandas DataFrame or Series, received {type(X)}"
+            raise TypeError(msg)
```

Comment:
would it be cleaner to use `is_pandas_dataframe` / `is_pandas_series`? (similarly for the polars code below)